File: src/test/kotlin/com/lhsystems/ListAppender.java
package com.lhsystems;

import ch.qos.logback.classic.spi.ILoggingEvent;
import ch.qos.logback.core.AppenderBase;

import java.util.ArrayList;
import java.util.List;

public class ListAppender extends AppenderBase<ILoggingEvent> {

    public final List<String> events = new ArrayList<>();

    @Override
    protected void append(ILoggingEvent event) {
        events.add(event.getFormattedMessage());
    }
}

File: src/test/kotlin/com/lhsystems/ListAppender.kt
package com.lhsystems

import ch.qos.logback.classic.spi.ILoggingEvent
import ch.qos.logback.core.AppenderBase

import java.util.ArrayList

class ListAppender : AppenderBase<ILoggingEvent>() {

    val events: MutableList<String> = ArrayList()

    override fun append(event: ILoggingEvent) {
        events.add(event.formattedMessage)
    }
}